src/comparator/field_comparator.py

```diff
@@ -257,6 +243,26 @@ def _compare_resource_reference(self, field_original, field_update):
                 resource_ref_update.source_code_line,
             )
 
+    def _resource_in_database(self, resource_ref) -> bool:
+        # Check whether the added resource reference is in the database.
+        if not self.global_resources_update:
+            return False
+        if self.field_update.child_type:
+            parent_resources = (
+                self.global_resources_update.get_parent_resources_by_child_type(
+                    resource_ref.value.child_type
+                )
+            )
+            if not parent_resources:
+                return False
+        else:
+            resources = self.global_resources_update.get_resource_by_type(
+                resource_ref.value.type
+            )
+            if not resources:
+                return False
+        return True
```

Comment:
Personal preference nit, feel free to ignore: I'm finding the control flow here a little hard to read. What do you think about
```python
if not self.global_resources_update:
    return False

resources = (
    self.global_resources_update.get_parent_resource_by_child_type(resource_ref.value.child_type)
    if self.field_update.child_type
    else self.global_resources_update.get_resource_by_type(resource_ref.value.type)
)

return bool(resources)
```
The lines are a little long, though. How much work would it be to shorten some of the names?

Reply:
Sure, this change looks reasonable to me, and for the name renaming, it is on my todo list, unfortunately the name spreads in several files, so I would prefer to make a separate PR for that, thanks!